### PR TITLE
Add initial secrets support

### LINKED ISSUE #283: Add podman secrets support
Currently it's missing

### PR DESCRIPTION
**Pull request by amenzhinsky**
Closes #283

### CODE REVIEW

File: plugins/modules/podman_secret.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    return 'no such secret' in err
+
+
+def podman_secret_create(module, executable, name, data, force):
+    if force:
+        rc, out, err = module.run_command([executable, 'secret', 'rm', name])
+        if rc == 0:
+            pass
+        elif not is_not_exists(err):
+            module.fail_json(msg="Unable to remove secret: %s" % err)
```
Comment (sshnaidm) [inline plugins/modules/podman_secret.py:90]: I think it's better not to fail here. If it doesn't exist - that's fine, we still want to create a secret.

File: plugins/modules/podman_secret.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    if rc == 0:
+        changed = True
+    elif is_not_exists(err):
+        pass
+    else:
+        module.fail_json(msg="Unable to remove secret: %s" % err)
+
+    return {
+        "changed": changed,
+        "id": out,
```
Comment (sshnaidm) [inline plugins/modules/podman_secret.py:116]: Should we return id? It doesn't exist anymore.

**Review by sshnaidm — CHANGES_REQUESTED** [review]
Thanks for contribution! Overall LGTM, a few comments inside. To run tests it will require a lil bit more code, I'll handle it in followup.

File: plugins/module_utils/podman/podman_container_lib.py [hunk 3]
```diff
@@ -106,6 +106,7 @@
     restart_policy=dict(type='str'),
     rm=dict(type='bool', aliases=['remove', 'auto_remove']),
     rootfs=dict(type='bool'),
+    secrets=dict(type='list', elements='str'),
```
Comment (sshnaidm) [inline plugins/module_utils/podman/podman_container_lib.py:109]: please add `no_log` here too

File: plugins/modules/podman_container.py [hunk 4]
```diff
@@ -595,6 +595,11 @@
       - If true, the first argument refers to an exploded container on the file
         system. The default is false.
     type: bool
+  secrets:
+    description:
+      - Add the named secrets into the container at /run/secrets.
+    type: list
+    elements: str
```
Comment (sshnaidm) [inline plugins/modules/podman_container.py:602]: not sure if `no_log` is needed here as well, but better to add

File: plugins/modules/podman_secret.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+options:
+  data:
+    description:
+      - The value of the secret. Required when C(state) is C(present).
+    type: str
+  executable:
+    description:
+      - Path to C(podman) executable if it is not in the C($PATH) on the
+        machine running C(podman)
+    default: 'podman'
```
Comment (sshnaidm) [inline plugins/modules/podman_secret.py:29]: please add `type: str`

**Review by sshnaidm — CHANGES_REQUESTED** [review]

File: plugins/modules/podman_container.py (hunk 4 above)
Comment (amenzhinsky) [reply plugins/modules/podman_container.py:602]: ERROR: plugins/modules/podman_container.py:0:0: invalid-documentation: DOCUMENTATION.options.secrets.no_log: extra keys not allowed @ data['options']['secrets']['no_log']. Got True

Comment (sshnaidm) [reply plugins/modules/podman_container.py:602]: Ok, fair enough

**Review by sshnaidm — APPROVED** [review]

File: plugins/modules/podman_secret.py [hunk 6]
```diff
@@ -78,43 +78,33 @@
...
     return {
         "changed": True,
-        "id": out,
     }
 
 
 def podman_secret_remove(module, executable, name):
     changed = False
-    rc, out, err = module.run_command([executable, 'secret', 'rm', name])
+    rc, _, err = module.run_command([executable, 'secret', 'rm', name])
```
Comment (sshnaidm) [inline plugins/modules/podman_secret.py:98]: pylint in ansible-test complains about "_": blacklisted-name: Black listed name "_"

**Review by sshnaidm — CHANGES_REQUESTED** [review]

**Review by sshnaidm — APPROVED** [review]

**Pull request merged**